lib/Fhp/FinTsNew.php

```diff
@@ -224,18 +224,23 @@ public function execute($action)
         /** @var HITANv6 $hitan */
         $hitan = $response->findSegment(HITANv6::class);
         if ($hitan !== null && $hitan->auftragsreferenz !== HITANv6::DUMMY_REFERENCE) {
-            if ($hitan->tanProzess !== 4) {
-                throw new UnexpectedResponseException("Unsupported TAN request type $hitan->tanProzess");
-            }
-            if ($this->bpd === null || $this->kundensystemId === null) {
-                throw new UnexpectedResponseException('Unexpected TAN request');
-            }
-            $action->setTanRequest($hitan);
-            if ($action instanceof DialogInitialization) {
-                $action->setDialogId($response->header->dialogId);
-                $action->setMessageNumber($this->messageNumber);
+
+            // After a TAN was send to the bank, the bank can include a HITAN Segment, with tanProzess 2
+            // and the Auftragsreferenz again
+
+            // This means, only the tanProzess 4 is actually a TAN request by the bank
+
+            if ($hitan->tanProzess === 4) {
```

Comment:
@Philipp91 fyi, die GLS macht das z.B. so

Reply:
Ich glaube nicht, dass diese Änderung hier nötig ist. Die DKB schickt auch ein `HKTAN` um zu sagen "Danke für die TAN, hat geklappt." Beispielsweise [hier](https://github.com/nemiah/phpFinTS/blob/c3366f5c10241d3301ba1a1019c3ffc84c528fa8/lib/Tests/Fhp/Integration/DKB/GetStatementOfAccountTest.php#L20) zu sehen.

Das "After a TAN was sent to the bank" dürfte gar nicht hier ausgeführt werden, das müsste unten in `submitTan()` passieren, oder? Und dort wird ja sogar explizit geprüft, ob tanProzess==2 zurückkommt.

Wenn du übrigens möchtest, dass Änderungen wie diese hier (möglicherweise), die die Kompatibilität mit deiner Bank herstellen, in Zukunft nicht versehentlich wieder über den Haufen geworfen werden, solltest du einen Integration Test hinzufügen.

Reply:
Hab es rausgenommen und gucke ob ich es vlt gar nicht brauche.